server/raft.go

```diff
@@ -1858,22 +1878,13 @@ func (n *raft) applyCommit(index uint64) error {
 		delete(n.acks, index)
 	}
 
-	// FIXME(dlc) - Can keep this in memory if this too slow.
-	ae, err := n.loadEntry(index)
-	if err != nil {
-		state := n.wal.State()
-		if index < state.FirstSeq {
-			return nil
-		}
-		if err != ErrStoreClosed && err != ErrStoreEOF {
-			if err == errBadMsg {
-				n.setWriteErrLocked(err)
-			}
-			n.warn("Got an error loading %d index: %v", index, err)
-		}
+	ae := n.pae[index]
+	if ae == nil {
+		n.warn("Could not load append entry %d", index)
 		n.commit = original
 		return errEntryLoadFailed
 	}
+	delete(n.pae, index)
```

Comment:
I experimented by adding a "deleted pae" map here (that is, keeping track of the index that we delete here) and when above we have `ae==nil` condition, do a lookup on that the deleted map and I found it. So it means that either there is a bug somewhere, or is it normal to have this function invoked with the same index.

Reply:
If we fail processing the entry trying to put the entry on the response channel maybe, will look.